modules/appeals_api/app/controllers/appeals_api/notice_of_disagreements/v0/notice_of_disagreements_controller.rb

```diff
@@ -30,13 +33,40 @@ class NoticeOfDisagreementsController < AppealsApi::V2::DecisionReviews::NoticeO
       ]
     }.freeze
 
+    def show
+      nod = AppealsApi::NoticeOfDisagreement.select(ALLOWED_COLUMNS).find(params[:id])
+      nod = with_status_simulation(nod) if status_requested_and_allowed?
+
+      render_notice_of_disagreement(nod)
+    rescue ActiveRecord::RecordNotFound
+      render_notice_of_disagreement_not_found(params[:id])
+    end
+
+    def create
+      nod = AppealsApi::NoticeOfDisagreement.new(
+        auth_headers: request_headers,
+        form_data: @json_body,
+        source: request_headers['X-Consumer-Username'].presence&.strip,
+        board_review_option: @json_body.dig('data', 'attributes', 'boardReviewOption'),
+        api_version: self.class::API_VERSION,
+        veteran_icn: @json_body.dig('data', 'attributes', 'veteran', 'icn')
+      )
+
+      return render_model_errors(nod) unless nod.validate
+
+      nod.save
+      AppealsApi::PdfSubmitJob.perform_async(nod.id, 'AppealsApi::NoticeOfDisagreement', 'v3')
+
+      render_notice_of_disagreement(nod)
+    end
+
     def download
       id = params[:id]
       notice_of_disagreement = AppealsApi::NoticeOfDisagreement.find(id)
 
       render_appeal_pdf_download(notice_of_disagreement, "#{FORM_NUMBER}-notice-of-disagreement-#{id}.pdf")
     rescue ActiveRecord::RecordNotFound
-      render_notice_of_disagreement_not_found
+      render_notice_of_disagreement_not_found(params[:id])
```

Comment:
Since `render_notice_of_disagreement_not_found` should also have access to `params`, you _could_ use `params[:id]` in that method directly, if you wanted to avoid passing a parameter. But that's just a difference in code style, so feel free to take or leave the suggestion.